core/runtime/wavm/module.cpp

```diff
@@ -37,6 +41,17 @@ namespace kagome::runtime::wavm {
       return nullptr;
     }
 
+    auto imports = WAVM::Runtime::getModuleIR(module).memories.imports;
+    if (imports.empty()) {
+      logger->critical("Memory type not found in imports.");
+      return nullptr;
+    }
+
+    module_params->intrinsicMemoryType = imports[0].type;
```

Comment:
It's really unintuitive that params are overwritten here instead of being read as params and I don't like. I think passing them as ModuleParams& instead of std::shared_ptr<ModuleParams> (why do we need an owning pointer to them here anyways?) would at least make it more obvious.